lib/article_status_manager.rb

```diff
@@ -34,62 +50,66 @@ def self.update_article_status(articles=nil)
-    update_article_ids deleted_ids
+    update_article_ids deleted_page_ids
 
     # Delete articles as appropriate
-    local_articles.where(id: deleted_ids).update_all(deleted: true)
-    local_articles.where(id: synced_ids).update_all(deleted: false)
-    ArticlesCourses.where(article_id: deleted_ids).destroy_all
-    limbo_revisions = Revision.where(article_id: deleted_ids)
-    RevisionImporter.move_or_delete_revisions limbo_revisions
+    articles.where(mw_page_id: deleted_page_ids).update_all(deleted: true)
+    articles.where(mw_page_id: synced_ids).update_all(deleted: false)
+    ArticlesCourses.where(article_id: deleted_page_ids).destroy_all
+    limbo_revisions = Revision.where(mw_page_id: deleted_page_ids)
+    RevisionImporter.new(@wiki).move_or_delete_revisions limbo_revisions
   end
 
   ##################
   # Helper methods #
   ##################
+  private
 
-  def self.update_title_and_namespace(synced_articles)
+  def update_title_and_namespace(synced_articles)
     # Update titles and namespaces based on ids (we trust ids!)
     synced_articles.map! do |sa|
       Article.new(
         id: sa['page_id'],
+        mw_page_id: sa['page_id'],
         title: sa['page_title'],
         namespace: sa['page_namespace'],
+        wiki_id: @wiki.id,
         deleted: false # Accounts for the case of undeleted articles
       )
     end
-    update_keys = [:title, :namespace, :deleted]
+    update_keys = [:title, :namespace, :deleted, :mw_page_id]
     Article.import synced_articles, on_duplicate_key_update: update_keys
   end
 
   # Check whether any deleted pages still exist with a different article_id.
   # If so, update the Article to use the new id.
-  def self.update_article_ids(deleted_ids)
-    maybe_deleted = Article.where(id: deleted_ids)
+  def update_article_ids(deleted_page_ids)
+    maybe_deleted = Article.where(mw_page_id: deleted_page_ids, wiki_id: @wiki.id)
 
     # These pages have titles that match Articles in our DB with deleted ids
     same_title_pages = Utils.chunk_requests(maybe_deleted, 100) do |block|
-      Replica.get_existing_articles_by_title block
+      Replica.new(@wiki).get_existing_articles_by_title block
     end
 
     # Update articles whose IDs have changed (keyed on title and namespace)
     same_title_pages.each do |stp|
-      resolve_article_id(stp, deleted_ids)
+      resolve_page_id(stp, deleted_page_ids)
     end
   end
 
-  def self.resolve_article_id(same_title_page, deleted_ids)
+  def resolve_page_id(same_title_page, deleted_page_ids)
     title = same_title_page['page_title']
     id = same_title_page['page_id']
     namespace = same_title_page['page_namespace']
 
     article = Article.find_by(
+      wiki_id: @wiki.id,
       title: title,
       namespace: namespace,
       deleted: false
     )
     return if article.nil?
-    return unless deleted_ids.include?(article.id)
+    return unless deleted_page_ids.include?(article.mw_page_id)
```

Comment:
Refers to article more than self (maybe move it to another class?) (FeatureEnvy)